src/library/scala/jdk/javaapi/FutureConverters.scala

```diff
@@ -66,11 +66,15 @@ object FutureConverters {
     * @return a Scala Future that represents the CompletionStage's completion
     */
   def asScala[T](cs: CompletionStage[T]): Future[T] = {
+    object Comfy {
+      // scala/bug#12918
+      def unapply(cf: CompletableFuture[T]) = Some(cf.toCompletableFuture)
```

Comment:
Will this create a `Some` object?

Reply:
Yes

Reply:
If we add some think like Akka's OptionVal we can then do it with allocation free.
Or we can just rewrite the this part of the code.

Reply:
What about 
```scala
        val cf = cs.toCompletableFuture
        if (cf.isDone && !cf.isCompletedExceptionally) {
          val p = new P[T](cs)
          p.tryComplete(Success(cf.join()))
          p.future
        } else {
          val p = new P[T](cs)
          cs.handle(p)
          p.future
        }
```

Reply:
Pushed a version without extractor.